tlsh.go

```diff
@@ -285,11 +340,31 @@ func HashBytes(blob []byte) (hash string, err error) {
 
 //Hash calculates the TLSH for the input file
 func Hash(filename string) (hash string, err error) {
-	f, err := os.Open(filename)
+	hash, err = hashFile(filename)
+	if err != nil {
+		return "", err
+	}
+
+	return hash, nil
+}
+
+// Diff calculate distance between two files
+func Diff(filenameA, filenameB string) (int, error) {
```

Comment:
Is there a reason Tlsh isn't exported? It seems like there are several different ways to use, so it might be best to keep all the functionality on main struct.

Is there any overlap between the current API and something like: https://golang.org/pkg/hash/#Hash

@glaslos what's your policy/thinking regarding **stable** APIs and versioning?